app/Services/ExportService.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+/**
+ * Provide useful methods to generate files to export.
+ */
+class FreshRSS_Export_Service {
+	/** @var string */
+	private $username;
+
+	/** @var FreshRSS_CategoryDAO */
+	private $category_dao;
+
+	/** @var FreshRSS_FeedDAO */
+	private $feed_dao;
+
+	/** @var FreshRSS_EntryDAO */
+	private $entry_dao;
+
+	/** @var FreshRSS_TagDAO */
+	private $tag_dao;
+
+	/**
+	 * Initialize the service for the given user.
+	 *
+	 * @param string $username
+	 */
+	public function __construct($username) {
+		$this->username = $username;
+
+		$this->category_dao = new FreshRSS_CategoryDAO($username);
+		$this->feed_dao = FreshRSS_Factory::createFeedDao($username);
+		$this->entry_dao = FreshRSS_Factory::createEntryDao($username);
+		$this->tag_dao = FreshRSS_Factory::createTagDao();
+	}
+
+	/**
+	 * Generate OPML file content.
+	 *
+	 * @return array First item is the filename, second item is the content
+	 */
+	public function generateOpml() {
+		require_once(LIB_PATH . '/lib_opml.php');
+
+		$view = new Minz_View();
+		$day = date('Y-m-d');
+		$categories = [];
+
+		foreach ($this->category_dao->listCategories() as $key => $category) {
+			$categories[$key]['name'] = $category->name();
+			$categories[$key]['feeds'] = $this->feed_dao->listByCategory($category->id());
+		}
+
+		$view->categories = $categories;
+
+		return [
+			"feeds_{$day}.opml.xml",
+			$view->helperToString('export/opml')
+		];
```

Comment:
I'm not completely happy with this pattern (i.e. returning an array containing the filename and the content) but it makes the things easier in the controller and the cli